source/solvers/heat_transfer.cc

```diff
@@ -969,7 +1019,780 @@ HeatTransfer<dim>::solve_linear_system(const bool initial_step,
+                      typename DoFHandler<dim>::active_cell_iterator cell_fd(
+                        &(*(this->triangulation)),
+                        cell->level(),
+                        cell->index(),
+                        dof_handler_fd);
+
+                      // Gather fluid dynamics information
+                      fe_face_values_fd.reinit(cell_fd, face);
+                      fe_face_values_fd[velocities].get_function_values(
+                        current_solution_fd, local_velocity_values);
+
+                      if (gather_vof)
+                        {
+                          // Get VOF active cell iterator
+                          typename DoFHandler<dim>::active_cell_iterator
+                            cell_vof(&(*(this->triangulation)),
+                                     cell->level(),
+                                     cell->index(),
+                                     dof_handler_vof);
+
+                          // Gather VOF information
+                          fe_face_values_vof->reinit(cell_vof, face);
+                          fe_face_values_vof->get_function_values(
+                            *this->multiphysics->get_solution(PhysicsID::VOF),
+                            phase_values);
+                        }
+
+                      // Loop on the quadrature points
+                      for (unsigned int q = 0; q < n_q_points_face; q++)
+                        {
+                          if (properties_manager.get_number_of_fluids() == 2)
+                            {
+                              // Blend the physical properties using the VOF
+                              // field
+                              thermal_conductivity = calculate_point_property(
+                                phase_values[q],
+                                thermal_conductivity_0[q],
+                                thermal_conductivity_1[q]);
+
+                              density =
+                                calculate_point_property(phase_values[q],
+                                                         density_0[q],
+                                                         density_1[q]);
+
+                              specific_heat =
+                                calculate_point_property(phase_values[q],
+                                                         specific_heat_0[q],
+                                                         specific_heat_1[q]);
+                            }
+
+                          normal_vector_ht =
+                            -fe_face_values_ht.normal_vector(q);
+
+                          heat_flux_bc +=
+                            (-thermal_conductivity * temperature_gradient[q] *
+                               normal_vector_ht +
+                             local_temperature_values[q] * density *
+                               specific_heat * local_velocity_values[q] *
+                               normal_vector_ht) *
```

Comment:
```suggestion
                          heat_flux_bc +=
                            (-thermal_conductivity * temperature_gradient[q] *
                               normal_vector_ht +
                             local_temperature_values[q] * density *
                               specific_heat * local_velocity_values[q] *
                               normal_vector_ht) *
```
So you calculate the full heat flux at the boundary? That's bit surprising, but it is true that will work also in the case of an outlet. I mean, it's more general this way, why not,